Samples/UnitySample/LifecyclePlugin.cs

```diff
@@ -0,0 +1,129 @@
+using Segment.Analytics;
+using Segment.Serialization;
+using UnityEngine;
+
+namespace UnitySample
+{
+    /// <summary>
+    /// Track your lifecycle events such as: installed, updated, opened, and backgrounded.
+    /// Copy and paste this class to your unity project, and add the following one-liner:
+    /// <code>
+    /// analytics.Add(new LifecyclePlugin());
+    /// </code>
+    /// Now your lifecycle events are automatically tracked.
+    /// </summary>
+    public class LifecyclePlugin : Plugin
+    {
+        public override PluginType Type => PluginType.Utility;
+
+        public override void Configure(Analytics analytics)
+        {
+            base.Configure(analytics);
+            LifecycleObserver.Instance.Analytics = analytics;
+        }
+
+
+        #region Observer Classes
+
+        /// <summary>
+        /// A singleton component that listens unity events and reports to analytics
+        /// </summary>
+        public class LifecycleObserver : Singleton<LifecycleObserver>
```

Comment:
yeah. singleton here is a trick that makes lifecycle object being auto added to every scene once it's loaded, so developer doesn't have to do anything extra. otherwise, the lifecycle object needs to be made as a prefab and developer has to drag it into every scene by hand, not a very good user experience.

I like the observer idea. in fact, I should make it function the way as its name stated. will update the pr